src/couchapps/WorkQueue/lists/workRestrictions.js

```diff
@@ -52,6 +52,16 @@ function(head, req) {
             var ele = row["doc"]["WMCore.WorkQueue.DataStructs.WorkQueueElement.WorkQueueElement"];
 
             // TODO: probably move this to a standalone function
+            // Don't check anything if useSiteListAsLocation/trusSiteLists is enabled
+            if (ele['NoLocationUpdate']) {
+                // subtract element jobs from site resources
+                if (first !== true) {
+                    send(",");
+                }
+                send(toJSON(row["doc"])); // need whole document, id etc...
+                first = false; // from now on prepend "," to output
+                break; // we have work, move to next element (break out of site loop)
+            }
```

Comment:
Since you are fixing it, maybe we should move the not related to site code to out side of the for loop. (line 52 - 74).
Also team and request checking ahead of if (ele['NoLocationUpdate'])